File: src/train_to_script.py
import json

def notebook_to_script(ipynb_filename, jl_filename):
    with open(ipynb_filename, 'r') as f:
        notebook = json.load(f)
        
    with open(jl_filename, 'w') as f:
        for cell in notebook['cells']:
            if cell['cell_type'] == 'code':
                f.write("#= Start of Cell =#\n")
                f.write(''.join(cell['source']))
                f.write("\n#= End of Cell =#\n\n")

ipynb_filename = "train.ipynb"
jl_filename = "train.jl"

notebook_to_script(ipynb_filename, jl_filename)

File: src/train_to_script.jl
using JSON

function notebook_to_script(ipynb_filename, jl_filename)
    data = JSON.parsefile(ipynb_filename)

    open(jl_filename, "w") do f
        for cell in data["cells"]
            if cell["cell_type"] == "code"
                println(f, "#= Start of Cell =#")
                println(f, join(cell["source"], "\n"))
                println(f, "#= End of Cell =#\n\n")
            end
        end
    end
end

notebook_to_script("./train.ipynb", "./train.jl")
